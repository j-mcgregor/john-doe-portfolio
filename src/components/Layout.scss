@import '../stylesheets/main.scss';

body {
    color: $default-color;
}
