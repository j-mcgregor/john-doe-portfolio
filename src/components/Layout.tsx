/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */
import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Header from './Header'

import 'flexboxgrid2'
import './Layout.css'

const Layout: React.SFC<{}> = ({ children }) => {
    const data = useStaticQuery(graphql`
        query SiteTitleQuery {
            site {
                siteMetadata {
                    title
                }
            }
        }
    `)
    return (
        <>
            <Header siteTitle={data.site.siteMetadata.title} />
            <div className="container">
                <div className="row">
                    <main>{children}</main>
                </div>
                <div className="row">
                    <div className="col-md-6">I'm on the left</div>
                    <div className="col-md-6">And I'm on the right</div>
                </div>
            </div>
            <footer className="container-fluid">
                Â© {new Date().getFullYear()}, Built with
                {` `}
                <a href="https://www.gatsbyjs.org">Gatsby</a>
            </footer>
        </>
    )
}
export default Layout
