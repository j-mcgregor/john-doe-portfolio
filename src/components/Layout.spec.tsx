import React from 'react'
import renderer from 'react-test-renderer'
import Layout from './Layout'

describe('Layout', () => {
    it('should render', () => {
        const layout = renderer.create(<Layout />)
        const layoutInstance = layout.toJSON()

        expect(layoutInstance).not.toBeNull()
        expect(layoutInstance).toHaveLength(3)
        if (layoutInstance) {
            expect(layoutInstance[0].type).toBe('nav')
            expect(layoutInstance[1].type).toBe('div')
            expect(layoutInstance[2].type).toBe('footer')
            expect(layoutInstance).toMatchSnapshot()
        }
    })
})
