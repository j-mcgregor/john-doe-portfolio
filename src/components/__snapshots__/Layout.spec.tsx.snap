// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="brand"
      href="/"
    >
      John Doe
    </a>
    <ul
      className="nav-list"
    >
      <li
        className="nav-item"
      >
        <a
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="container-fluid"
  />,
  <footer
    className="footer"
  >
    Reach me on
    <div
      className="social"
    />
  </footer>,
]
`;
